engine/src/main/java/org/hibernate/validator/internal/engine/constraintvalidation/HibernateConstraintValidatorInitializationContextImpl.java

```diff
@@ -59,4 +62,43 @@ public Duration getTemporalValidationTolerance() {
 			return null;
 		}
 	}
+
+	@Override
+	public boolean equals(Object o) {
+		if ( this == o ) {
+			return true;
+		}
+		if ( o == null || getClass() != o.getClass() ) {
+			return false;
+		}
+
+		HibernateConstraintValidatorInitializationContextImpl hibernateConstraintValidatorInitializationContextImpl = (HibernateConstraintValidatorInitializationContextImpl) o;
+
+		if ( scriptEvaluatorFactory != null ? scriptEvaluatorFactory != hibernateConstraintValidatorInitializationContextImpl.scriptEvaluatorFactory : hibernateConstraintValidatorInitializationContextImpl.scriptEvaluatorFactory != null ) {
+			return false;
+		}
+		if ( clockProvider != null ? clockProvider != hibernateConstraintValidatorInitializationContextImpl.clockProvider : hibernateConstraintValidatorInitializationContextImpl.clockProvider != null ) {
+			return false;
+		}
+		if ( temporalValidationTolerance != null ? !temporalValidationTolerance.equals( hibernateConstraintValidatorInitializationContextImpl.temporalValidationTolerance ) : hibernateConstraintValidatorInitializationContextImpl.temporalValidationTolerance != null ) {
+			return false;
+		}
+		if ( constraintValidatorPayload != null ? constraintValidatorPayload != hibernateConstraintValidatorInitializationContextImpl.constraintValidatorPayload : hibernateConstraintValidatorInitializationContextImpl.constraintValidatorPayload != null ) {
```

Comment:
As `scriptEvaluatorFactory` `clockProvider` and `temporalValidationTolerance` cannot be `null` we can skip the null checks for them, and also we do a `!=` check anyway.

Reply:
@marko-bekhta Updated, please check the `equals` method again (I squashed my fix in the original commit to keep a nice commit history)